ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala

```diff
@@ -1464,6 +1559,126 @@ private class JdbcLedgerDao(
+      updateLedgerEnd(newLedgerEnd, externalOffset)
+      Try({
+        SQL(queries.SQL_INSERT_PACKAGE_UPLOAD_ENTRY)
+          .on(
+            "ledger_offset" -> offset,
+            "recorded_at" -> entry.recordTime,
+            "submission_id" -> entry.submissionId,
+            "participant_id" -> entry.participantId,
+            "typ" -> typ,
+            "rejection_reason" -> reasonOrNull(entry)
+          )
+          .execute()
+        PersistenceResponse.Ok
+      }).recover {
+        case NonFatal(e) if e.getMessage.contains(queries.DUPLICATE_KEY_ERROR) =>
+          logger.warn(
+            s"Ignoring duplicate package upload entry submission for submissionId ${entry.submissionId} participantId ${entry.participantId}")
+          conn.rollback()
+          PersistenceResponse.Duplicate
+      }.get
+    }
+  }
+
+  override def lookupPartyAllocationEntry(
+      submissionId: SubmissionId): Future[Option[PartyAllocationLedgerEntry]] = {
+    dbDispatcher
+      .executeSql("load_party_allocation_entry", None) { implicit conn =>
+        SQL_SELECT_PARTY_ALLOCATION_ENTRY
+          .on("submissionId" -> submissionId)
+          .as(partyAllocationEntryParser.*)
+          .headOption
+          .map(_._2)
+      }
+  }
+
+  override def lookupPackageUploadEntry(
+      submissionId: SubmissionId): Future[Option[PackageUploadLedgerEntry]] = {
+    dbDispatcher
+      .executeSql("load_package_upload_entry", None) { implicit conn =>
+        SQL_SELECT_PACKAGE_UPLOAD_ENTRY
+          .on("submissionId" -> submissionId)
+          .as(packageUploadEntryParser.*)
+          .headOption
+          .map(_._2)
+      }
+  }
+
+  override def storePartyAllocationEntry(
+      offset: LedgerOffset,
+      newLedgerEnd: LedgerOffset,
+      externalOffset: Option[ExternalOffset],
+      submissionId: SubmissionId,
+      participantId: ParticipantId,
+      entry: PartyAllocationLedgerEntry): Future[PersistenceResponse] = {
+    val typ = entry match {
+      case PartyAllocationLedgerEntry.Accepted(_, _, _, _) => entryAcceptType
+      case _ => entryRejectType
+    }
+
+    dbDispatcher.executeSql("store_party_allocation_entry") { implicit conn =>
+      updateLedgerEnd(newLedgerEnd, externalOffset)
+      Try({
+        SQL(queries.SQL_INSERT_PARTY_ALLOCATION_ENTRY)
+          .on(
+            "ledger_offset" -> offset,
+            "recorded_at" -> entry.recordTime,
+            "submission_id" -> submissionId,
+            "participant_id" -> participantId,
+            "party" -> optionalPartyDetails(entry).map(_.party.toString).orNull,
+            "display_name" -> optionalPartyDetails(entry).map(_.displayName.orNull),
+            "typ" -> typ,
+            "rejection_reason" -> partyAllocationReasonOrNull(entry)
+          )
+          .execute()
+        PersistenceResponse.Ok
+      }).recover {
+        case NonFatal(e) if e.getMessage.contains(queries.DUPLICATE_KEY_ERROR) =>
```

Comment:
I don't think the insert query results in a duplicate error being thrown (seeing that it contains `on conflict do nothing`).

Reply:
resolved - just show full error now and removed the guard -- should never get here but if we do it will give a more descriptive error